connman/plugins/sailfish_wifi.c

```diff
@@ -638,7 +643,23 @@ static void wifi_bss_free(struct wifi_bss *bss_data)
 
 static inline gboolean wifi_bss_weak(const struct wifi_bss *bss_data)
 {
-	return (bss_data->signal < WIFI_WEAK_RSSI);
+	int weak_rssi = WIFI_WEAK_RSSI;
+	int services;
+
+	services = connman_service_get_service_count(CONNMAN_SERVICE_TYPE_WIFI);
+	if (services) {
+		int multiplier = (int)floor(
+					services / WIFI_SERVICE_COUNT_INTERVAL);
```

Comment:
Isn't that equivalent to just `multiplier = services / WIFI_SERVICE_COUNT_INTERVAL`?

Reply:
I may have been overthinking here, again, and as a part of older approach I tried. Simplified with 227e5a58c5a82057284f23d68820568f467a864f